rust-engine/src/names.rs

```diff
@@ -261,7 +216,12 @@ pub mod codegen {
                     format!("{data}__{}", item.disambiguator)
                 }
             }))
-            .chain(if matches!(&id.kind, DefKind::Ctor { .. }) {
+            .chain(if explicit_id.is_constructor {
+                Some("Constructor".to_string())
+            } else {
+                None
+            })
+            .chain(if matches!(id.kind, DefKind::Ctor(..)) {
```

Comment:
So this now contains both "Constructor" and "ctor"? Seems redundant?

Reply:
Yes, it is redundant, it's a wierdness: this Ctor fragment is inserted by Rustc only on tuple structs, but not on anything else :(
I can add a TODO

Reply:
I filled https://github.com/cryspen/hax/issues/1657 and added a comment, thanks!